### PR TITLE
enhance(schedule): add next run field to schedule

### PR DESCRIPTION
**Pull request by wsan3**
## Description
Add a "next run" field to the schedule object so that we can display when a schedule is going to run NEXT.

## Desired View
Screenshot 2024-05-15 at 10 06 04 AM

## Related
https://github.com/go-vela/community/issues/871

### CODE REVIEW

File: database/schedule/list_active.go [hunk 1]
```diff
@@ -4,7 +4,9 @@ package schedule
 
 import (
 	"context"
+	"time"
 
+	"github.com/adhocore/gronx"
```
[the remark this answers is not in the document]
Comment (wsan3) [reply database/schedule/list_active.go:9]: Ran `make lintfix`

File: database/schedule/get.go [hunk 2]
```diff
@@ -4,7 +4,9 @@ package schedule
 
 import (
 	"context"
+	"time"
 
+	"github.com/adhocore/gronx"
```
[the remark this answers is not in the document]
Comment (wsan3) [reply database/schedule/get.go:9]: Ran `make lintfix`

File: api/types/schedule.go [hunk 3]
```diff
@@ -336,6 +350,19 @@ func (s *Schedule) SetError(err string) {
 	s.Error = &err
 }
 
+// SetNextRun sets the NextRun field.
+//
+// When the provided Schedule type is nil, it
+// will set nothing and immediately return.
+func (s *Schedule) SetNextRun(err int64) {
```
Comment (plyr4) [inline api/types/schedule.go:357]: confused about the var name `err int64` is it an error?

File: database/types/schedule.go [hunk 4]
```diff
@@ -99,6 +103,7 @@ func (s *Schedule) ToAPI() *api.Schedule {
 	schedule.SetScheduledAt(s.ScheduledAt.Int64)
 	schedule.SetBranch(s.Branch.String)
 	schedule.SetError(s.Error.String)
+	schedule.SetNextRun(nextTime.Unix())
```
Comment (plyr4) [inline database/types/schedule.go:106]: if `gronx.NextTickAfter` returns err does this line panic?

File: api/types/schedule.go (hunk 3 above)
Comment (wsan3) [reply api/types/schedule.go:357]: Good catch!

File: database/types/schedule.go (hunk 4 above)
Comment (plyr4) [reply database/types/schedule.go:106]: ah okay cool. good question, i guess we should set it to zero in the event of an error to indicate theres no valid next_run_time?

Comment (wsan3) [reply database/types/schedule.go:106]: Yeah we can do that

**Review by plyr4 — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]

**Pull request merged**